packages/kbn-server-route-repository-client/src/create_repository_client.ts

```diff
@@ -10,28 +10,52 @@ import type { CoreSetup, CoreStart } from '@kbn/core-lifecycle-browser';
 import {
   RouteRepositoryClient,
   ServerRouteRepository,
-  DefaultClientOptions,
   formatRequest,
 } from '@kbn/server-route-repository-utils';
+import { httpResponseIntoObservable } from '@kbn/sse-utils-client';
+import { from } from 'rxjs';
+import { HttpFetchOptions, HttpFetchQuery, HttpResponse } from '@kbn/core-http-browser';
+import { omit } from 'lodash';
 
 export function createRepositoryClient<
   TRepository extends ServerRouteRepository,
-  TClientOptions extends Record<string, any> = DefaultClientOptions
->(core: CoreStart | CoreSetup) {
+  TClientOptions extends HttpFetchOptions = {}
+>(core: CoreStart | CoreSetup): RouteRepositoryClient<TRepository, TClientOptions> {
+  const fetch = (
+    endpoint: string,
+    params: { path?: Record<string, string>; body?: unknown; query?: HttpFetchQuery } | undefined,
+    options: TClientOptions
+  ) => {
+    const { method, pathname, version } = formatRequest(endpoint, params?.path);
+
+    return core.http[method](pathname, {
+      ...options,
+      body: params && params.body ? JSON.stringify(params.body) : undefined,
+      query: params?.query,
+      version,
+    });
+  };
+
   return {
-    fetch: (endpoint, optionsWithParams) => {
-      const { params, ...options } = (optionsWithParams ?? { params: {} }) as unknown as {
-        params?: Partial<Record<string, any>>;
-      };
+    fetch: (endpoint, ...args) => {
+      const allOptions = args[0] ?? {};
+      const params = 'params' in allOptions ? (allOptions.params as Record<string, any>) : {};
+      const otherOptions = omit(allOptions, 'params') as TClientOptions;
 
-      const { method, pathname, version } = formatRequest(endpoint, params?.path);
+      return fetch(endpoint, params, otherOptions) as any;
+    },
+    stream: (endpoint, ...args) => {
```

Comment:
Could you update the README in the server package with an example of using streaming as well?

Reply:
✅